colbert/indexing/codecs/residual_embeddings.py

```diff
@@ -119,17 +123,22 @@ def save(self, path_prefix):
 
     def lookup_codes(self, pids):
         assert self.mmap_index
-        codes = torch.zeros((sum(self.doclens[pid] for pid in pids]))
+        codes = torch.zeros(sum([self.pid_to_chunk_metadata[pid][1] for pid in pids]))
+
         pids_per_chunk = defaultdict(list)
         for pid in pids:
-            chunk_idx = self.pid_to_chunk_idx[pid]
+            chunk_idx = self.pid_to_chunk_metadata[pid][0]
             pids_per_chunk[chunk_idx].append(pid)
+
         offset = 0
-        for chunk_idx in sorted(chunks.keys()):
+        for chunk_idx in sorted(pids_per_chunk.keys()):
```

Comment:
This is a problem in the code I wrote, but one concern I have is that it may be possible for the pids passed in to `lookup_pids` to not be in sorted order - if that is the case, then iterate over the chunks in sorted order would produce an output that's inconsistent with what the calling function expects

Reply:
I guess the solution would be to iterate over the pids in the original order, then find the chunk each pid belongs to and extract the relevant data from that chunk
Or alternatively, we could sort the pids upstream somewhere (assuming this is not done already)